### ISSUE TITLE
Data fields not preserved on import/export, in BSLightingShaderProperty 

### ISSUE THREAD

**Issue by dev-nll** [BadDogSkyrim/PyNifly#179]
Various fields in BSLightingShaderProperty don't get always preserved in the final mesh after importing into Blender followed by exporting back to mesh.

Not all BSLightingShaderProperty have this problem, only the highlighted ones:

![image](https://github.com/BadDogSkyrim/PyNifly/assets/1366642/217d23a9-8551-4c44-a00c-219ef487d89e)

These are the non-preserved data fields:

![image](https://github.com/BadDogSkyrim/PyNifly/assets/1366642/3d1129bd-b08e-4f4a-b855-fdcac2f7677f)

Original mesh prior to Blender import/export:

[pinus04_599919f0passthru_lod.zip](https://github.com/BadDogSkyrim/PyNifly/files/13022905/pinus04_599919f0passthru_lod.zip)

**Comment by BadDogSkyrim** [comment]
I'm about to play with shaders. I'll include this stuff when I do.

**Comment by dev-nll** [comment]
> I'm about to play with shaders. I'll include this stuff when I do.

One more thing: The shape sometimes doesn't get preserved when importing/exporting; the ".001" at the end is deleted sometimes. I've attached a mesh where this happens. This is fairly important since DynDOLOD requires correct shape names in this data field.

[mesh.zip](https://github.com/BadDogSkyrim/PyNifly/files/13037019/mesh.zip)

![image](https://github.com/BadDogSkyrim/PyNifly/assets/1366642/8feceeef-60cc-4936-8b06-eff6094bb31b)

Also, this is off topic, but it would be extremely useful if default export settings/options are preserved each time we save something. Although I'm not sure how easy that would be to implement, just thought I'd throw that out there.

**Comment by BadDogSkyrim** [comment]
Huh. So there are meshes out there in the wild that have .001 extensions in the name?

The .00n extensions are generally added by Blender to keep object names unique. In pynifly on export I've been assuming any such extensions were added by Blender and were not part of the original nif mesh, so I've been stripping them.

**Comment by dev-nll** [comment]
> Huh. So there are meshes out there in the wild that have .001 extensions in the name?
> 
> The .00n extensions are generally added by Blender to keep object names unique. In pynifly on export I've been assuming any such extensions were added by Blender and were not part of the original nif mesh, so I've been stripping them.

Yeah, all of the original meshes in Nature of the Wild Lands, like this:

![image](https://github.com/BadDogSkyrim/PyNifly/assets/1366642/4a564932-05ea-4214-83d8-d28c4ba62be2)

When I export from PyNifly, it will delete the suffix ".*" (e.g. ".001") if and only if that suffix exists on the first BSTriShape. Any such suffixes on the 2nd, 3rd, .. BSTriShapes will be left alone.

Ideally, for my application, all suffixes are left alone, since they exist in the original model. Is there a way to detect whether the suffix already exists when the mesh is first imported, and if it already exists, do not delete it when it gets exported?

**Comment by dev-nll** [comment]
I'm running into the same issues with V13.0.0.

I've attached an example. Upon export, the name "picea_8(1).001" changes to "picea_8(1)", and Texture Clamp Mode "CLAMP_S_CLAMP_T" changes to "WRAP_S_WRAP_T".

[test.zip](https://github.com/BadDogSkyrim/PyNifly/files/13258886/test.zip)

**Comment by podvornyakva** [comment]
Maby it is cause of multiply import, where the same material is present? Only new scene and one object of a type import. Or do handwork to change material to duplicated meshes... either disable material copying in blender.

**Comment by BadDogSkyrim** [comment]
I'm guessing this is just a bug. I was a bit lazy with shader attributes the first time around. I just revamped the whole scheme and could have gotten something wrong.

@podvornyakva, the problem is when a nif already has .001 at the end of a name. pyNifly thinks it's irrelevant and strips it, so the exported name doesn't match the import.

**Comment by podvornyakva** [comment]
> the problem is when a nif already has .001
Probably it is a bug. Bethesda's or other moder. I've never come to that kind of "behavior".

**Comment by dev-nll** [comment]
> I'm guessing this is just a bug. I was a bit lazy with shader attributes the first time around. I just revamped the whole scheme and could have gotten something wrong.

To clarify, it's not a newly introduced bug in v13.0.0, It's the same behavior as with the previous version. I just wanted to bring it to your attention, that's all.

**Comment by podvornyakva** [comment]
> It's from this mod: [NOTWL](https://www.nexusmods.com/skyrimspecialedition/mods/63604)

Curiously that i'm also try to lay my hands on that stuff. But meshes of that mod do not imports with pynifly...

**Comment by dev-nll** [comment]
> > It's from this mod: [NOTWL](https://www.nexusmods.com/skyrimspecialedition/mods/63604)
> 
> Curiously that i'm also try to lay my hands on that stuff. But meshes of that mod do not imports with pynifly...

Try my example above, it comes from NOTWL and works. Also, try deleting the parts of the mesh you're not working on (e.g. collisions) and then import only the part you're working on (e.g. trunk). If none of this works, there's some else going wrong with your installation or use of PyNifly.

**Comment by podvornyakva** [comment]
> Try my example above, it comes from NOTWL and works.

Nevermind. My fault, BSA browser stripped away all files while unpack. After running WINE with .NET prefix all come to normal.

**Comment by BadDogSkyrim** [comment]
Should be fixed now. Let me know if there are still issues.
